src/clj/fluree/db/json_ld/policy/enforce.cljc

```diff
@@ -52,6 +52,19 @@
     (get-in policy-map [const/iri-modify :default])
     (get-in policy-map [const/iri-view :default])))
 
+(defn policies-for-flake
+  [{:keys [policy namespace-codes] :as db} [s p o :as flake] modify?]
+  (let [s-iri (iri/sid->iri s namespace-codes)
+        p-iri (iri/sid->iri p namespace-codes)
+        o-val (if (iri/sid? o) (iri/sid->iri o namespace-codes) o) ]
```

Comment:
This, and the code below is extremely performance sensitive... it could be done millions of times in a single query. This work is irrelevant if these new types of policies are not present.

We definitely need a way to skip all of this extra work preemptively. At minimum a flag on the policy could be checked, but checking the flag is still potentially extra work... but seems like a reasonable short-term solution. I think a composed policy function that only composes the relevant checks per flake that a given policy enforces feels like a more efficient way, but could be something we do later.